### PR TITLE
Use NewGVN and remove special-cases for phi nodes in PatchPeepholeOpt

### LINKED ISSUE #945: LLPC fails assertion: `isEqual(getLiveRegsBefore(*NonDbgMI, *LIS), LRS)'
Found using a build with assertions ON and sanitizers ON, using these commit hashes:

/vulkandriver/drivers/AMDVLK
88d3e8e911055fe9c6a7ae1960b9fae13a70a9cf
/vulkandriver/drivers/llpc
aacd74a46f62e652b63cfb2b6f145f6e489d74d4
/vulkandriver/drivers/llvm-project
ccbc834ac4f92e63a9a003bd5a18cb9fcf3c6121
/vulkandriver/drivers/pal
9074bd0bdaf67fa84399dfabed3ec437951838f8
/vulkandriver/drivers/spvgen
c054813a9a894e32aaaa04c6717a667c15f60cfd
/vulkandriver/drivers/third_party/cwpack
7387247eb9889ddcabbc1053b9c2052e253b088e
/vulkandriver/drivers/third_party/metrohash
712f76fee75d69b23a1ea8f6465752c3ccaaf9a2
/vulkandriver/drivers/xgl
598c6832a4983f5b75b38a589fca5be80a2f3bb0

Assertions are required to trigger the issue; sanitizers should not be required.

[Files needed to reproduce the issue](https://github.com/GPUOpen-Drivers/llpc/files/5167444/tofile.zip)

Command to reproduce the issue:

```
amdllpc -gfxip=9.0.0 -verify-ir -auto-layout-desc tofile/shader.frag.spv -o temp.out
```

I see this output:

```
amdllpc: /vulkandriver/drivers/llvm-project/llvm/lib/Target/AMDGPU/GCNSchedStrategy.cpp:469: void llvm::GCNScheduleDAGMILive::computeBlockPressure(const llvm::MachineBasicBlock *): Assertion `isEqual(getLiveRegsBefore(*NonDbgMI, *LIS), LRS)' failed.
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace.
Stack dump:
0.	Program arguments: /data/temp/llpc-from-docker/amdllpc -gfxip=9.0.0 -verify-ir -auto-layout-desc shader.frag.spv 
1.	Running pass 'CallGraph Pass Manager' on module 'lgcPipeline'.
2.	Running pass 'Machine Instruction Scheduler' on function '@_amdgpu_ps_main'
```

Issue found by the [GraphicsFuzz](https://github.com/google/graphicsfuzz) project.

### PR DESCRIPTION
**Pull request by LLJJDD**
Address #957. Done some local testing, NewGVN should be able to do the work PatchPeepholeOpt used to do.

This seems to fix #945 and #946 unexpectedly.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 877a3d55d3429a9700924fc389da924e294f9ce4
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev a52f7c12f9ce5aca5c0c7f799f72b1b3362121bc</li>
<li>SPVGEN: dev c054813a9a894e32aaaa04c6717a667c15f60cfd</li>
<li>XGL: dev feb87bf013f2b4bd16f7a1da39a5a97fdaa7470d</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 240f3dd769edd6615fa478ecb9f7a761922ac026</li>
</ul></details>

<details><summary>CTS tests (Failed: 1/131664)</summary><ul><li>Built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 22797/43895 (51.9%)</li>
<li>Failed: 0/43895 (0.0%)</li>
<li>Skipped: 21098/43895 (48.1%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 22795/43895 (51.9%)</li>
<li><details><summary>Failed: 1/43895 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.image.mutable.2d.r32_sint_r8g8b8a8_unorm_clear_load
Stack trace: Flake
```
</p></details></li>

<li>Skipped: 21099/43895 (48.1%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 21622/43874 (49.3%)</li>
<li>Failed: 0/43874 (0.0%)</li>
<li>Skipped: 22252/43874 (50.7%)</li>
</ul></details>


</ul></details>

File: lgc/patch/PatchPeepholeOpt.cpp [hunk 1]
```diff
@@ -569,94 +569,6 @@ void PatchPeepholeOpt::visitPHINode(PHINode &phiNode) {
 
     return;
   }
-
```
Comment (ruiling) [inline lgc/patch/PatchPeepholeOpt.cpp:572]: I think it is better the deletion goes to a separate patch, that would be easy for bisect or revert if we find something wrong.

Comment (LLJJDD) [reply lgc/patch/PatchPeepholeOpt.cpp:572]: Done, split the change into two commits.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 9d4d97d8999554dc5c6624a857bc73f8f3dacfb5
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev a52f7c12f9ce5aca5c0c7f799f72b1b3362121bc</li>
<li>SPVGEN: dev c054813a9a894e32aaaa04c6717a667c15f60cfd</li>
<li>XGL: dev feb87bf013f2b4bd16f7a1da39a5a97fdaa7470d</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev ff1e992e7e6cd35b73564ac9d5fa3cf4e351b9d1</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/131664)</summary><ul><li>Built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 22797/43895 (51.9%)</li>
<li>Failed: 0/43895 (0.0%)</li>
<li>Skipped: 21098/43895 (48.1%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 22796/43895 (51.9%)</li>
<li>Failed: 0/43895 (0.0%)</li>
<li>Skipped: 21099/43895 (48.1%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 21622/43874 (49.3%)</li>
<li>Failed: 0/43874 (0.0%)</li>
<li>Skipped: 22252/43874 (50.7%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 561f0e1fbcd35d1105defed1c8f5cccae67a0f5f
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev a52f7c12f9ce5aca5c0c7f799f72b1b3362121bc</li>
<li>SPVGEN: dev c054813a9a894e32aaaa04c6717a667c15f60cfd</li>
<li>XGL: dev feb87bf013f2b4bd16f7a1da39a5a97fdaa7470d</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev ff1e992e7e6cd35b73564ac9d5fa3cf4e351b9d1</li>
</ul></details>

<details><summary>CTS tests (Failed: 2/131664)</summary><ul><li>Built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 22797/43895 (51.9%)</li>
<li>Failed: 0/43895 (0.0%)</li>
<li>Skipped: 21098/43895 (48.1%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 22794/43895 (51.9%)</li>
<li><details><summary>Failed: 2/43895 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.draw.instanced.draw_indexed_indirect_vk_primitive_topology_triangle_fan_attrib_divisor_1
Stack trace: Flake

FAILURE: dEQP-VK.subgroups.arithmetic.compute.subgroupmax_uvec4
Stack trace: Flake
```
</p></details></li>

<li>Skipped: 21099/43895 (48.1%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 21622/43874 (49.3%)</li>
<li>Failed: 0/43874 (0.0%)</li>
<li>Skipped: 22252/43874 (50.7%)</li>
</ul></details>


</ul></details>

**Comment by ruiling** [comment]
looks good to me. I am not sure @dstutt do you have more comments?

**Comment by dstutt** [comment]
I've made the same change locally (but hadn't progressed my testing enough), so I think this change is probably a good one.
Have you done any analysis on performance? That's the main one I had left to do.

**Comment by LLJJDD** [comment]
> I've made the same change locally (but hadn't progressed my testing enough), so I think this change is probably a good one.
> Have you done any analysis on performance? That's the main one I had left to do.

I've run an internal benchmark locally and observe no noticeable performance impact.

**Comment by dstutt** [comment]
> > I've made the same change locally (but hadn't progressed my testing enough), so I think this change is probably a good one.
> > Have you done any analysis on performance? That's the main one I had left to do.
> 
> I've run an internal benchmark locally and observe no noticeable performance impact.

Great, LGTM then.

**Review by dstutt — APPROVED** [review]
LGTM

**Comment by dstutt** [comment]
I notice that the internal testing hasn't passed though - maybe a retest? (Looks like the failures were a couple of flake tests on gfx9)

**Comment by JaxLinAMD** [comment]
retest this please (just in case)

**Pull request merged**